src/queue.rs

```diff
@@ -131,7 +132,8 @@ pub struct DescriptorChain<M: GuestAddressSpace> {
     mem: M::T,
     desc_table: GuestAddress,
     queue_size: u16,
-    ttl: u16, // used to prevent infinite chain cycles
+    ttl: u16,   // used to prevent infinite chain cycles
+    index: u16, // descriptor index of the chain header
```

Comment:
Looks a bit clearer if we rename this and the accessor method to something like `head_index`.